### PR TITLE
[SID-1145] <DynamicFlow> component

### PR DESCRIPTION
**Pull request by ikovic**
## Description

[YouTrack issue](https://todo.irdcorp.dev/issue/SID-1145)

Implemented the `DynamicFlow` component that allows performing login with the authentication method that depends on the user handle.

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have generated the new version of the docs website and smoke tested it
- [ ] I have checked that my changes haven't caused semantic errors in the existing docs
- [ ] I have updated the README and DEVELOPMENT files

### CODE REVIEW

File: packages/react/src/components/dynamic-flow/initial.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        <ConfiguredHandleForm handleSubmit={handleSubmit} />
+      </FormProvider>
+      {shouldRenderDivider && <Divider>{text["initial.divider"]}</Divider>}
+      <Oidc
+        providers={oidcFactors}
+        handleClick={(factor) =>
+          flowState.logIn({
+            factor,
+            handle: undefined,
+          })
```
Comment (jakewhelan) [inline packages/react/src/components/dynamic-flow/initial.tsx:46]: How should middleware be implemented here?

File: packages/react/src/dev.tsx [hunk 2]
```diff
@@ -67,13 +66,72 @@ function Config() {
...
+    options: {
+      provider: "google",
+      client_id: import.meta.env.VITE_GOOGLE_SSO_CLIENT_ID,
+    },
+  },
+  {
+    method: "oidc",
+    options: {
+      provider: "azuread",
+      client_id: import.meta.env.VITE_GOOGLE_SSO_CLIENT_ID,
```
Comment (jakewhelan) [inline packages/react/src/dev.tsx:104]: Is this a mistake?

File: packages/react/src/dev.tsx [hunk 3]
```diff
@@ -67,13 +66,72 @@ function Config() {
...
+    },
+  },
+];
+
+const ConfiguredDynamicFlow = () => {
+  return (
+    <ConfigurationProvider text={{ "initial.oidc": "Continue with" }}>
+      <DynamicFlow
+        className="formWrapper"
+        getFactor={getFactor}
```
Comment (jakewhelan) [inline packages/react/src/dev.tsx:114]: Does this mean its only possible to have a single factor with `DynamicFlow`?

Is this component a drop-in replacement for `<Form>`?

**Review by jakewhelan — APPROVED** [review]
-

File: packages/react/src/components/dynamic-flow/initial.tsx (hunk 1 above)
Comment (jake-slashid) [inline packages/react/src/components/dynamic-flow/initial.tsx:46]: How should middleware be implemented here?

File: packages/react/src/dev.tsx (hunk 2 above)
Comment (jake-slashid) [inline packages/react/src/dev.tsx:104]: Is this a mistake?

**Review by jake-slashid — APPROVED** [review]
I don't deeply understand whats going on here but nothing looks alarming

I have some clarifying questions for my own benefit, but no feedback

**Review by ahmed-gz — APPROVED** [review]
LGTM, left minor comments!

**Pull request merged**